store/rootmulti/store.go

```diff
@@ -253,7 +253,7 @@ func (rs *Store) loadVersion(ver int64, upgrades *types.StoreUpgrades) error {
 
 		store, err := rs.loadCommitStoreFromParams(key, commitID, storeParams)
 		if err != nil {
-			return errorsmod.Wrap(err, "failed to load store")
+			return errorsmod.Wrapf(err, "failed to load store for %s", key.Name())
```

Comment:
This change is not needed for the fix but it adds some very valuable debug output